docs/adr/0001-create-research-template-docker-repo.md

```diff
@@ -0,0 +1,25 @@
+# 1. Create the research-template-docker repo
+
+Date: 2024-06-04
+
+## Status
+
+Accepted
+
+## Context
+
+We are using [dev containers](https://code.visualstudio.com/docs/devcontainers/create-dev-container) to provide a configuration for Codespaces that gives researchers a useful development environment to use while working on their OpenSAFELY projects. We have created a custom Docker image for this dev container environment. This custom image requires a Dockerfile and other supporting files to be created.
+
+## Decision
+
+We will create a separate GitHub repo to store the Dockerfile and other supporting files. Specifically, we will store the following in the research-template-docker repo:
+
+* The Dockerfile used to generate the dev containers Docker image
+* The tooling and GitHub actions to build the Docker image and to run the tests
+* Some of the supporting files for the research-template (e.g documentation and ADRs).
```

Comment:
But _why_? For example:

> We will store the Dockerfile in research-template-docker because we do not want researchers to modify it and because we want to version it.

Reply:
I've tried to address that in the consequences section, which I've reworded in my most recent commit. I've tried to keep the decision section short and to just state the decision without the reasoning.